modules/scratchFetch.ts

```diff
@@ -0,0 +1,242 @@
+import { defineEvent } from "strife.js";
+import { getSettings } from "./settings.js";
+import constants from "../common/constants.js";
+
+defineEvent("messageCreate", async (message) => {
+	const notSet = getSettings(message.author, false)?.scratchEmbeds === undefined;
+	if (!getSettings(message.author).scratchEmbeds) {
+		return;
+	}
+	const scratchUrlRegex =
+		/(?<!<)https?:\/\/scratch\.mit\.edu\/(projects|users|studios|discuss)\/[/-\w]+\/?(?!>)/; //gpt wrote the regex and like half of this code
+	const match = message.content.match(scratchUrlRegex);
+
+	if (!match) {
+		return;
+	}
+
+	const urlParts = match[0].split("/");
+	const type = urlParts[3];
+
+	async function fetchApiData(apiUrl: string): Promise<any> {
+		try {
+			const response = await fetch(apiUrl);
+
+			const data = await response.json();
+			return data;
+		} catch (error) {}
+	}
+	function long(text: string, maxLength: number, appendText: string): string {
+		if (text.length > maxLength) {
+			return text.substr(0, maxLength) + appendText;
+		}
+		return text;
+	}
+	switch (type) {
+		case "projects":
+			const projectdata = await fetchApiData(
+				`https://${constants.urls.scratchApi}/projects/${urlParts[4]}/`,
+			).catch(() => {});
+
+			message.reply({
+				embeds: [
+					{
+						title: projectdata.title,
+						description: `**desc**: ${projectdata.description}\n**inst**: ${projectdata.instructions}`,
+						color: 0x885cd4,
+
+						fields: [
+							{
+								name: `views`,
+								value: projectdata.stats.views,
+								inline: true,
+							},
+							{
+								name: `loves`,
+								value: projectdata.stats.loves,
+								inline: true,
+							},
+							{
+								name: `favorites`,
+								value: projectdata.stats.favorites,
+								inline: true,
+							},
+							{
+								name: `remixes`,
+								value: projectdata.stats.remixes,
+								inline: true,
+							},
+						],
```

Comment:
let's put these in the same order as they are on scratch
loves, favorites, remixes, views

also there's 4 of these, so lets put them in 2 rows of 2. we have to do some trickery to do that but basically keep them all `inline` and after the 2nd one add another one with both a name and a value of `constants.zeroWidthSpace`

finally, add a is remix? field that is only shown when it's true. put it after `views`